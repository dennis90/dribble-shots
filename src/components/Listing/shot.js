import React from 'react'

import ShotFooter from './shot-footer'


const Shot = ({shot}) => {
  const getStyle = () => ({
      backgroundImage: `url(${shot.images.normal})`
  })

  return (
    <div className="shot" style={getStyle()}>
      <ShotFooter shot={shot} />
    </div>
  )
}

export default Shot
