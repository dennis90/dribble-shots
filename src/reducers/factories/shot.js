import {Factory} from 'rosie'

export const ShotFactory = new Factory()
  .sequence('id')
  .sequence('title', index => `Shot #${index}`)
  .sequence('description', index => `Shot Description #${index}`)
  .attrs({
    images: {
      hidpi: 'https://dribbble/images/hidpi',
      normal: 'https://dribbble/images/normal',
    },
    views_count: 100,
    likes_count: 100,
  })
  .attr('user', ['user'], user => UserFactory.build(user))

export const UserFactory = new Factory()
  .sequence('name', index => `User #${index}`)
  .attrs({
    avatar_url: 'https://dribbble.com/user/avatar.png',
    html_url: 'https://dribbble.com/user/',
  })
