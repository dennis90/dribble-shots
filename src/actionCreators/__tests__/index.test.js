import fetchMock from 'fetch-mock'

import types from '../../actionTypes'
import {fetchDribbleShots} from '../index'
import {ShotFactory} from '../../reducers/factories/shot'

describe('Action Creators', () => {
  it('`fetchDribbleShots` action creator', async () => {
    const dispatch = jest.fn()
    const foundShots = ShotFactory.buildList(3)
    fetchMock.get('*', foundShots)
    await fetchDribbleShots()(dispatch)
    const [path, request] = fetchMock.lastCall()

    expect(path).toMatchSnapshot('Request URL')
    expect(request).toMatchSnapshot('Request Header')
    expect(dispatch.mock.calls).toMatchSnapshot('Action dispatches')
    fetchMock.restore()
  })
})
