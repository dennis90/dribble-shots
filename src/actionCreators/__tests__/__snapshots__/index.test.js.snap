// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action dispatches 1`] = `
Array [
  Array [
    Object {
      "payload": Array [
        Object {
          "description": "Shot Description #1",
          "id": 1,
          "images": Object {
            "hidpi": "https://dribbble/images/hidpi",
            "normal": "https://dribbble/images/normal",
          },
          "likes_count": 100,
          "title": "Shot #1",
          "user": Object {
            "avatar_url": "https://dribbble.com/user/avatar.png",
            "html_url": "https://dribbble.com/user/",
            "name": "User #1",
          },
          "views_count": 100,
        },
        Object {
          "description": "Shot Description #2",
          "id": 2,
          "images": Object {
            "hidpi": "https://dribbble/images/hidpi",
            "normal": "https://dribbble/images/normal",
          },
          "likes_count": 100,
          "title": "Shot #2",
          "user": Object {
            "avatar_url": "https://dribbble.com/user/avatar.png",
            "html_url": "https://dribbble.com/user/",
            "name": "User #2",
          },
          "views_count": 100,
        },
        Object {
          "description": "Shot Description #3",
          "id": 3,
          "images": Object {
            "hidpi": "https://dribbble/images/hidpi",
            "normal": "https://dribbble/images/normal",
          },
          "likes_count": 100,
          "title": "Shot #3",
          "user": Object {
            "avatar_url": "https://dribbble.com/user/avatar.png",
            "html_url": "https://dribbble.com/user/",
            "name": "User #3",
          },
          "views_count": 100,
        },
      ],
      "type": "SHOTS_FETCHED",
    },
  ],
]
`;

exports[`Request Header 1`] = `
Object {
  "headers": Headers {
    "_headers": Object {
      "authorization": Array [
        "Bearer replace-with-a-valid-key",
      ],
    },
  },
}
`;

exports[`Request URL 1`] = `"https://api.dribbble.com/v1/shots/"`;
